src/plots/layout_attributes.js

```diff
@@ -125,6 +125,14 @@ module.exports = {
                 'Padding is muted if the respective anchor value is *middle*/*center*.'
             ].join(' ')
         }),
+        automargin: {
+            valType: 'boolean',
+            dflt: false,
+            editType: 'plot',
+            description: [
+                'Determines whether the title can automatically push the figure margins.'
```

Comment:
Could you please improve the description to mention the role of other attributes here?
Thank you!

Reply:
How do we feel about this: 

 > Determines whether the title can automatically push the figure margins. If yref=paper then the margin will expand to ensure that the title doesn’t overlap with the edges of the container. If yref=container then the margins will also ensure that the title doesn’t overlap with the plot area. If automargin=true and the margins need to be expanded, then y will be set to a default 1 and yanchor will be set to an appropriate default to ensure that minimal margin space is needed. Note that when yref=paper, only 1 or 0 are allowed y values. Invalid values will be reset to the default 1. 

cc @alexcjohnson

Reply:
Excellent! Just a small edit in the 3rd sentence, did I get this right?

> Determines whether the title can automatically push the figure margins. If yref=paper then the margin will expand to ensure that the title doesn’t overlap with the edges of the container. If yref=container then the margins will ensure that the title doesn’t overlap with the plot area, tick labels, and axis titles. If automargin=true and the margins need to be expanded, then y will be set to a default 1 and yanchor will be set to an appropriate default to ensure that minimal margin space is needed. Note that when yref=paper, only 1 or 0 are allowed y values. Invalid values will be reset to the default 1.

Reply:
Yes, thanks for adding that!